compiler/resolution/wrappers.cpp

```diff
@@ -1266,13 +1266,7 @@ static void errorIfValueCoercionToRef(CallExpr* call, ArgSymbol* formal) {
   }
 }
 
-static bool isUnmanagedClass(Type* t) {
-  if (DecoratedClassType* dt = toDecoratedClassType(t))
-    if (dt->isUnmanaged())
-      return true;
-
-  return false;
-}
+// Can this be combined with type.cpp:isBorrowedClass()?  Which is right?
```

Comment:
Hi @mppf: Low priority, but: in looking for ways to check whether something was a class, I found the routines here and in functionResolution.cpp and thought they'd be more natural in type.cpp alongside the various isClass() style queries.  The two overloads of isUnmanagedClass() seemed identical, but the two versions of isBorrowedClass() were not, so I moved one of them arbitrarily before noticing the second.  Should one of them "win" over the other, or do both need to exist?

Reply:
I'm missing something - I don't see two versions of isBorrowedClass, could you help me out with some line numbers?

Reply:
There's the one just below this comment (called `isBorrowClass()`... I forgot that their names didn't quite match), and the other was at functionResolution.cpp:5789, but in this PR has moved to type.cpp:1071.

Reply:
Right, so we had

```
static bool isBorrowClass(Type* t) {
  if (DecoratedClassType* dt = toDecoratedClassType(t)) {
    if (dt->isUnmanaged())
      return false;
    else
      return true;
  } else if (isClass(t)) {
    return true;
  }

  return false;
}
```

and also:

```
// Should this be public, like isNilableClassType() ?
static bool isBorrowedClass(Type* t) {
  if (isClass(t))
    return true; // borrowed, non-nilable

  if (DecoratedClassType* dt = toDecoratedClassType(t))
    return dt->isBorrowed();

  return false;
}
```

The first above is erroneous (and was probably good enough for the one call in wrappers.cpp - it used to be correct but then we got more DecoratedClassTypes) and the second is the one to put into type.cpp.

Reply:
Great, thanks!